base_report_to_printer/models/ir_actions_report.py

```diff
@@ -41,12 +41,19 @@ def onchange_printing_printer_id(self):
         self.printer_tray_id = False
 
     @api.model
-    def print_action_for_report_name(self, report_name):
+    def _get_report_from_action_id(self, action_id):
+        report_obj = self.env['ir.actions.report']
+        conditions = [('id', '=', action_id)]
+        context = self.env['res.users'].context_get()
+        return report_obj.with_context(context).sudo().search(conditions, limit=1)
+
+    @api.model
+    def print_action_for_report_name(self, action_id):
         """Returns if the action is a direct print or pdf
 
         Called from js
         """
-        report = self._get_report_from_name(report_name)
```

Comment:
This always get the fist `ir.actions.report`. It means that if we want to have multiple report types to be printed is not possible.